ikasaneip/developer/mvn-archetype/ikasan-standalone-filesystem-im/src/main/resources/archetype-resources/jar/src/test/java/com/ikasan/sample/spring/boot/builderpattern/ApplicationTest.java

```diff
@@ -42,62 +42,149 @@
 
 import org.apache.activemq.junit.EmbeddedActiveMQBroker;
 import org.ikasan.builder.IkasanApplication;
+import org.ikasan.component.endpoint.filesystem.producer.FileProducer;
+import org.ikasan.component.endpoint.filesystem.producer.FileProducerConfiguration;
+import org.ikasan.component.endpoint.jms.spring.consumer.JmsContainerConsumer;
+import org.ikasan.component.endpoint.jms.spring.consumer.SpringMessageConsumerConfiguration;
+import org.ikasan.spec.configuration.ConfigurationManagement;
+import org.ikasan.spec.configuration.ConfiguredResource;
 import org.ikasan.spec.flow.Flow;
 import org.ikasan.spec.module.Module;
+import org.ikasan.testharness.flow.jms.MessageListenerVerifier;
+import org.ikasan.testharness.flow.rule.IkasanFlowTestRule;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.springframework.aop.framework.Advised;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.SocketUtils;
 
+import javax.annotation.Resource;
+import javax.jms.TextMessage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.util.List;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
 /**
  * This test class supports the <code>SimpleExample</code> class.
- * 
+ *
  * @author Ikasan Development Team
  */
 @RunWith(SpringRunner.class)
+@SpringBootTest(classes = {Application.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class ApplicationTest
 {
-    public EmbeddedActiveMQBroker broker = new EmbeddedActiveMQBroker();
+    private static String FILE_PRODUCER_FILE_NAME = "testProducer.out";
+
+    private static String FILE_CONSUMER_FILE_NAME = "testConsumer.txt";
+
+    @Resource
+    private EmbeddedActiveMQBroker broker;
+
+    @Resource
+    private Module moduleUnderTest;
+
+    @Resource
+    private ConfigurationManagement configurationManagement;
+
+    @Resource
+    MessageListenerVerifier messageListenerVerifierTarget;
 
-    @Before
-    public void setup()
+    @Resource
+    JmsTemplate jmsTemplate;
+
+    @After public void shutdown() throws IOException
     {
-        broker.start();
+        Files.deleteIfExists(FileSystems.getDefault().getPath(FILE_PRODUCER_FILE_NAME));
+        Files.deleteIfExists(FileSystems.getDefault().getPath(FILE_CONSUMER_FILE_NAME));
```

Comment:
Duplicate line?

Reply:
One is producer file one is consumer file

Reply:
ah ok - my bad :)